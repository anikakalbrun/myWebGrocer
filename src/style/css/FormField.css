label {
    font-weight: 200;
    font-size: 16px;
}

.error {
    color: #ff4002
}


div.has-error {
    background-color: #ff4002;
    width: 100%;
    height: 3px;
    margin: 0;
    padding: 0;
}

.no-error {
    height: 25px;
}

.form-horizontal.form-group label {
    color: #222222;
}

.form-control {
    background-color: #fff;
    border: 1px solid #333333;
    border-radius: 2px;
    font-size: 14px;
    height: 38px;
}

input.valid {
    background-image: url('../images/check.png');
    background-repeat: no-repeat;
    background-position: 98% 50%;
    background-size: 20px;
}
