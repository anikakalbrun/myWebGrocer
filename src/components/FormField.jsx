import React from 'react';
import {object, string, bool, shape} from 'prop-types';

import FieldWrapper from './FieldWrapper';
import '../style/css/FormField.css';

FormField.propTypes = {
    input: object,
    label: string,
    placeholder: string,
    autoComplete: string,
    disabled: bool,
    meta: shape({
        touched: bool,
        error: string
    })
};

FormField.defaultProps = {
    type: 'text',
    disabled: false,
    autoComplete: "off"
};

export default function FormField({input, type, label, placeholder, disabled, autoComplete, meta: {touched, error}}) {
    return (
        <FieldWrapper label={label} touched={touched} error={error}>
            <input {...input}
                   placeholder={placeholder}
                   type={type}
                   disabled={disabled}
                   className={`form-control ${touched && !error ? 'valid' : ''}`}
                   autoComplete={autoComplete}/>
        </FieldWrapper>
    )
}