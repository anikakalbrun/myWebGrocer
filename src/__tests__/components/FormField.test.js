import React from 'react';
import FormField from '../../components/FormField';

let wrapper;
let props = {
    input: {},
    label: "First Name",
    placeholder: "First Name",
    autoComplete: "given-name",
    disabled: false,
    meta: {
        touched: false,
        error: undefined
    }
};

describe('FormField', () => {

    it('renders the FieldWrapper component', () => {
        wrapper = shallow(<FormField {...props}/>);
        expect(wrapper.find('FieldWrapper').length).toEqual(1);
    });

    it('renders input as child component and provides propper props to it', () => {
        wrapper = shallow(<FormField {...props}/>);
        expect(wrapper.find('FieldWrapper').at(0).props().children).toMatchSnapshot();
    });

    beforeEach(() => {
        props = {
            input: {},
            label: "First Name",
            placeholder: "First Name",
            autoComplete: "given-name",
            disabled: false,
            meta: {
                touched: true,
                error: 'required'
            }
        };

        wrapper = shallow(<FormField {...props}/>);
    });

    it('provides error prop to the FieldWrapper component', () => {
        expect(wrapper.find('FieldWrapper').at(0).props().error).toEqual('required');
    });

    it('provides label prop to the FieldWrapper component', () => {
        expect(wrapper.find('FieldWrapper').at(0).props().label).toEqual('First Name');
    });

    it('provides touched prop to the FieldWrapper component', () => {
        expect(wrapper.find('FieldWrapper').at(0).props().touched).toEqual(true);
    });

})
;
