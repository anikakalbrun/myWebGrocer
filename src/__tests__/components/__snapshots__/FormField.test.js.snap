// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders input as child component and provides propper props to it 1`] = `
<input
  autoComplete="given-name"
  className="form-control "
  disabled={false}
  placeholder="First Name"
  type="text"
/>
`;
